tests/connect_redirect/connect_redirect_tests.cpp

```diff
@@ -296,15 +315,15 @@ _load_and_attach_ebpf_programs(_Outptr_ struct bpf_object** return_object)
 
     if (_globals.attach_v6_program) {
         printf("Attaching v6 program\n");
-        bpf_program* connect_program_v6 = bpf_object__find_program_by_name(object, "connect_redirect6");
+        bpf_program* connect_program_v6 = bpf_object__find_program_by_name(unique_object.get(), "connect_redirect6");
         REQUIRE(connect_program_v6 != nullptr);
 
         result = bpf_prog_attach(
             bpf_program__fd(const_cast<const bpf_program*>(connect_program_v6)), 0, BPF_CGROUP_INET6_CONNECT, 0);
         REQUIRE(result == 0);
     }
 
-    *return_object = object;
+    *return_object = unique_object.get();
```

Comment:
won't this close the object when unique_object goes out of scope on return?
Seems like you should need to do unique_object.reset(nullptr) to take ownership?